test/e2e/prerender-crawler.test.ts

```diff
@@ -0,0 +1,97 @@
+import { createNext } from 'e2e-utils'
+import { renderViaHTTP } from 'next-test-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+
+describe('Prerender crawler handling', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files: {
+        'pages/index.js': `
+          export default function Page() { 
+            return <p>index page</p>
+          } 
+        `,
+        'pages/blog/[slug].js': `
+          import {useRouter} from 'next/router'
+          
+          export default function Page({ slug }) { 
+            const router = useRouter()
+            
+            if (router.isFallback) {
+              return 'Loading...'
+            }
+            
+            return (
+              <>
+                <p id='page'>slug page</p>
+                <p id='slug'>{slug}</p>
+              </>
+            )
+          } 
+          
+          export async function getStaticProps({ params }) {
+            return {
+              props: {
+                slug: params.slug
+              }
+            }
+          } 
+          
+          export async function getStaticPaths() {
+            return {
+              paths: ['/blog/first'],
+              fallback: true
+            }
+          }
+        `,
+      },
+    })
+  })
+  afterAll(() => next.destroy())
+
+  it('should return prerendered page for correctly', async () => {
+    const html = await renderViaHTTP(next.url, '/blog/first')
+    expect(html).not.toContain('Loading...')
+    expect(html).toContain('first')
+    expect(html).toContain('slug page')
+  })
+
+  it('should return fallback for non-crawler correctly', async () => {
+    const html = await renderViaHTTP(next.url, '/blog/test-1')
+    expect(html).toContain('Loading...')
+  })
```

Comment:
Probably add a test for a browser user-agent to confirm it returns the fallback

Reply:
Updated to use a browser user-agent here